bats/tests/helpers/vm.bash

```diff
@@ -65,6 +67,25 @@ start_kubernetes() {
         --kubernetes-version "$RD_KUBERNETES_PREV_VERSION"
 }
 
+start_application() {
+    start_kubernetes
+    wait_for_apiserver
+
+    # the docker context "rancher-desktop" may not have been written
+    # even though the apiserver is already running
+    if using_docker; then
+        wait_for_container_engine
```

Comment:
Let's do this in a separate PR.

Also needs an updated explanation; the comment right now explains that the docker context might not be created yet, even though the apiserver is already accessible. Since we don't define a docker context when using `containerd`, this wait it not necessary.

FWIW, I would much prefer if we would set up the context much earlier and not wait until dockerd is already running.